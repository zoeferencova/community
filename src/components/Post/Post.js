import React, { Component } from "react";
import { withRouter } from "react-router-dom";
import { PropTypes } from 'prop-types';
import CommUnityContext from "../../contexts/context";
import { ButtonLight, ProfilePicture } from "../Utils/Utils";
import Task from "../Task/Task";
import UserDataService from "../../services/user-data-service";
import styles from "./Post.module.css";

class Post extends Component {
    static contextType = CommUnityContext;

    render() {
        return (   
            <div className={styles.post}>
                <div className={styles.details}>
                    <div>
                        <div className={styles.postHeader}>
                            <div className={styles.nameSection}>
                                <ProfilePicture first_name={this.props.first_name} user_name={this.context.user.first_name} />
                                <h4>{this.props.first_name} {this.props.post_type === "offer" ? "offered to help" : "requested help"}</h4>
                            </div>
                            <div>
                                <span>{this.props.post_type === "offer" ? <i className={`fas fa-heart ${styles.heart}`}></i> : <i className={`fas fa-hand-paper ${styles.hand}`}></i>}</span>
                                <span className={styles.distance}>{UserDataService.metersToMiles(this.props.distance_from_user) + " mi"}</span>
                            </div>
                        </div>
                        <ul className={styles.helpItems}>
                            {this.props.categories.map(task => <Task key={task} task={task} />)}
                        </ul>
                        <p className={styles.description}>{this.props.description}</p>
                    </div>
                </div>
                <ButtonLight type="button" onClick={() => this.props.history.push(`/post/${this.props.id}`)} className={styles.buttonLink}>{this.props.post_type === "offer" ? "Accept offer" : "Offer to help"}</ButtonLight>
            </div>        
        )
    }
}

export default withRouter(Post);

Post.propTypes = {
    first_name: PropTypes.string,
    post_type: PropTypes.string,
    distance_from_user: PropTypes.string,
    categories: PropTypes.array,
    description: PropTypes.string,
    id: PropTypes.number
}