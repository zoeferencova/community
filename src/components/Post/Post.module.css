.post {
    display: flex;
    flex-direction: column;
    width: 90%;
    padding: 20px;
    text-align: left;
    margin: 15px 0px;
    background-color: white;
    box-shadow: 0 1px 3px 0 rgba(0,0,0,.1),0 1px 2px 0 rgba(0,0,0,.06);
    border-radius: 10px;
}

.postHeader {
    display: flex;
    flex-direction: row;
    align-items: center;
    margin: 0 0 20px 0;

    height: 30px;
}

.postHeader h4 {
    margin-left: 10px;
}

.details {
    display: flex;
    justify-content: space-between;
}

.distance {
    font-size: 14px;
    font-weight: bold;
    letter-spacing: 1px;
    color: #404040;
}

.button, .buttonLink {
    margin-top: 15px;
    align-self: center;
    width: 200px;
    font-size: 14px;
    border: none;
    background-color: #f2f4f7;
}

.buttonLink:hover {
    border: none;
    background-color: #e4e9f0;
    transition: 0.5s;
    opacity: 1;
}

.helpItems {
    display: flex;
    list-style: none;
    padding-left: 0px;
}

.high, .medium, .low {
    font-size: 12px;
    display: inline-block;
    padding-top: 8px;
}

.high i, .medium i, .low i {
    font-size: 10px;
    margin-right: 3px;
}

.high {
    color: #ad3434;
}

.medium {
    color: #edcb64;
}

.low {
    color: #3d9145;
}

.heart, .hand {
    margin-right: 10px;
}

.heart {
    color: #bf5e6b;
}

.hand {
    color: #e6c78e;
}

@media only screen and (min-width: 830px) {
    .post {
        width: 70%;
    }
}