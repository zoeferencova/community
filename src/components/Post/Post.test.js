import React from "react";
import { shallow, mount, render } from "enzyme";
import Post from "./Post";
import { BrowserRouter } from "react-router-dom";
import CommUnityContext from "../../contexts/context";

it("expect to render Post component", () => {
    const mockValue = {"user":{"id":1,"first_name":"Zoe","email":"zoe@gmail.com","location":{"lat":40.748225,"lng":-73.89698620000001},"radius":"3.00"},"user_posts":[{"id":38,"user_id":1,"post_type":"request","description":null,"urgency":"low","date_created":"2020-09-23T20:33:01.817Z","location":{"lat":40.748225,"lng":-73.89698620000001},"radius":"3.00","first_name":"Zoe","categories":["Phone call","Running errands"],"distance_from_user":"0"},{"id":40,"user_id":1,"post_type":"request","description":null,"urgency":"low","date_created":"2020-09-23T20:33:19.724Z","location":{"lat":40.748225,"lng":-73.89698620000001},"radius":"3.00","first_name":"Zoe","categories":["Online chat","Phone call"],"distance_from_user":"0"}],"neighborhood_posts":[{"id":4,"user_id":2,"post_type":"offer","description":null,"urgency":null,"date_created":"2020-06-11T21:24:05.882Z","location":{"lat":40.7532952,"lng":-73.9068973},"radius":"3.00","first_name":"James","categories":["Running errands","Picking up supplies"],"distance_from_user":"1007"},{"id":17,"user_id":3,"post_type":"request","description":null,"urgency":"medium","date_created":"2020-09-16T21:09:57.136Z","location":{"lat":40.7432931,"lng":-73.9229366},"radius":"2.75","first_name":"Anna","categories":["Dog walking","Phone call","Picking up supplies"],"distance_from_user":"2254"},{"id":18,"user_id":4,"post_type":"offer","description":null,"urgency":null,"date_created":"2020-09-16T21:12:26.077Z","location":{"lat":40.737975,"lng":-73.8801301},"radius":"4.00","first_name":"Yasmin","categories":["Dog walking","Picking up supplies"],"distance_from_user":"1821"},{"id":19,"user_id":11,"post_type":"request","description":null,"urgency":"high","date_created":"2020-09-18T16:51:24.339Z","location":{"lat":40.7532952,"lng":-73.9068973},"radius":"4.00","first_name":"Demo","categories":["Phone call","Running errands"],"distance_from_user":"1007"}],"chats":[{"id":25,"user1":{"first_name":"Zoe","id":1},"user2":{"first_name":"James","id":2},"post":{"id":4,"user_id":2,"post_type":"offer","description":null,"urgency":null,"date_created":"2020-06-11T21:24:05.882246"},"messages":[{"id":285,"chat_id":25,"sender_id":1,"message_timestamp":"2020-09-17T18:22:47.088676","message_content":"Hi James, I'm also based in Queens and I need help getting my medication from the pharmacy while I'm quarantining. Are you near the Duane Reade on Roosevelt Ave in Jackson Heights?"},{"id":286,"chat_id":25,"sender_id":2,"message_timestamp":"2020-09-17T18:23:29.967525","message_content":"Hey Zoe! Nice to meet you. Yes, I'm about half a mile from there."},{"id":287,"chat_id":25,"sender_id":2,"message_timestamp":"2020-09-17T18:23:45.977949","message_content":"Happy to pick up whatever you need. Groceries, medication, anything!"},{"id":288,"chat_id":25,"sender_id":1,"message_timestamp":"2020-09-17T18:24:12.017073","message_content":"Awesome, thank you so much! "},{"id":289,"chat_id":25,"sender_id":1,"message_timestamp":"2020-09-17T18:24:45.266031","message_content":"Would it be ok if I sent you a list of a couple of things? I think there's a grocery store right by that Duane Reade and a couple of pantry items would be a lifesaver."},{"id":290,"chat_id":25,"sender_id":2,"message_timestamp":"2020-09-17T18:25:06.813435","message_content":"That works for me! What do you need? :) "},{"id":320,"chat_id":25,"sender_id":1,"message_timestamp":"2020-09-24T02:09:43.209634","message_content":"hi"}]}],"socket":{"io":{"nsps":{},"subs":[{},{},{},{},{},{}],"opts":{"path":"/socket.io","hostname":"evening-springs-65156.herokuapp.com","secure":true,"port":"443"},"_reconnection":true,"_reconnectionAttempts":null,"_reconnectionDelay":1000,"_reconnectionDelayMax":5000,"_randomizationFactor":0.5,"backoff":{"ms":1000,"max":5000,"factor":2,"jitter":0.5,"attempts":0},"_timeout":20000,"readyState":"open","uri":"https://evening-springs-65156.herokuapp.com","connecting":[null],"lastPing":null,"encoding":false,"packetBuffer":[],"encoder":{},"decoder":{"reconstructor":null,"_callbacks":{"$decoded":[null]}},"autoConnect":true,"engine":{"secure":true,"agent":false,"hostname":"evening-springs-65156.herokuapp.com","port":"443","query":{},"upgrade":true,"path":"/socket.io/","forceJSONP":false,"jsonp":true,"forceBase64":false,"enablesXDR":false,"withCredentials":true,"timestampParam":"t","transports":["polling","websocket"],"transportOptions":{},"readyState":"open","writeBuffer":[],"prevBufferLen":0,"policyPort":843,"rememberUpgrade":false,"binaryType":null,"perMessageDeflate":{"threshold":1024},"pfx":null,"key":null,"passphrase":null,"cert":null,"ca":null,"ciphers":null,"rejectUnauthorized":true,"forceNode":false,"isReactNative":false,"id":"SfG92F29xpI4v0ukAAAE","upgrades":["websocket"],"pingInterval":25000,"pingTimeout":5000,"pingIntervalTimer":9,"pingTimeoutTimer":24,"transport":{"usingBrowserWebSocket":true,"path":"/socket.io/","hostname":"evening-springs-65156.herokuapp.com","port":"443","secure":true,"query":{"EIO":3,"transport":"websocket","sid":"SfG92F29xpI4v0ukAAAE"},"timestampParam":"t","readyState":"open","agent":false,"enablesXDR":false,"withCredentials":true,"pfx":null,"key":null,"passphrase":null,"cert":null,"ca":null,"ciphers":null,"rejectUnauthorized":true,"forceNode":false,"isReactNative":false,"_callbacks":{"$drain":[null],"$packet":[null],"$error":[null],"$close":[null]},"ws":{},"writable":true},"_callbacks":{"$data":[null],"$ping":[null],"$pong":[null],"$error":[null],"$close":[null],"$heartbeat":[null]},"upgrading":false},"skipReconnect":false,"_callbacks":{"$open":[null],"$packet":[null],"$close":[null]}},"nsp":"/","ids":0,"acks":{},"receiveBuffer":[],"sendBuffer":[],"connected":true,"disconnected":false,"flags":{},"subs":[{},{},{}],"_callbacks":{"$connecting":[null],"$connect":[null,null],"$reconnect":[null],"$PRIVATE_MESSAGE":[null],"$NEW_CHAT":[null],"$CHAT_TO_REMOVE":[null]},"id":"SfG92F29xpI4v0ukAAAE"},"activeChat":null,"timeZone":"America/New_York","loading":false}
    expect(
        render(
            <CommUnityContext.Provider value={mockValue}>
                <BrowserRouter>
                    <Post categories={mockValue.neighborhood_posts[0].categories} first_name={mockValue.neighborhood_posts[0].first_name} />
                </BrowserRouter>
            </CommUnityContext.Provider>
        )
    ).toMatchSnapshot()
})
