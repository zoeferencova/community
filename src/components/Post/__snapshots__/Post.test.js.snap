// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Post component 1`] = `
<div
  class="post"
>
  <div
    class="details"
  >
    <div>
      <div
        class="postHeader"
      >
        <div
          class="nameSection"
        >
          <div
            class="circle indigo undefined"
            first_name="James"
            user_name="Zoe"
          >
            J
          </div>
          <h4>
            James requested help
          </h4>
        </div>
        <div>
          <span>
            <i
              class="fas fa-hand-paper hand"
            />
          </span>
          <span
            class="distance"
          >
            NaN mi
          </span>
        </div>
      </div>
      <ul
        class="helpItems"
      >
        <li
          class="task errands"
        >
          Running errands
        </li>
        <li
          class="task supplies"
        >
          Picking up supplies
        </li>
      </ul>
      <p
        class="description"
      />
    </div>
  </div>
  <button
    class="button buttonLink"
    type="button"
  >
    Offer to help
  </button>
</div>
`;
